File: authdemo/src/main/java/com/example/authdemo/service/dto/AccountDto.java
package com.example.authdemo.service.dto;


import com.example.authdemo.model.Role;
import lombok.Data;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Set;

@Data
public class AccountDto {

    private Long id;

    @NotEmpty(message = "firstName is mandatory")
    private String firstName;


    @NotEmpty(message = "lastName is mandatory")
    private String lastName;


    //@NotNull(message = "user must be contains one of more roles")
    private Set<Role> roles;


    @NotEmpty(message = "email is mandatory")
    @Email(regexp = "^(.+)@(.+)$", message = "email is not valid")
    private String email;


    @NotEmpty(message = "password is mandatory")
    private String password;

}

File: ChatClient/ChatClient/model/dto/AccountDTO.cs
namespace ChatClient.model.dto
{
    internal class AccountDTO
    {
        private string _FirstName;

        private string _LastName;

        public string FirstName
        {
            get
            {
                return _FirstName;
            }
            set
            {
                if (value != null)
                {
                    _FirstName = value;
                }
                else
                {
                    throw new InvalidInputException("firstName field is mandatory");
                }
            }
        }

        public string LastName
        {
            get
            {
                return _LastName;
            }
            set
            {
                if (value != null)
                {
                    _LastName = value;
                }
                else
                {
                    throw new InvalidInputException("lastName field is mandatory");
                }
            }
        }

        public AccountDTO(string firstName, string lastName)
        {
            FirstName = firstName;
            LastName = lastName;
        }
    }
}